File: Data Structures & Algorithms/house-robber/submission-1.java
class Solution {
    public int rob(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }

        if (nums.length == 1) {
            return nums[0];
        }

        int prev_2 = nums[0];
        int prev_1 = Math.max(nums[1], nums[0]);

        for (int i = 2; i < nums.length; i++) {
            int current = Math.max(nums[i] + prev_2, prev_1);
            prev_2 = prev_1;
            prev_1 = current;
        }

        return prev_1;
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-1.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        dp = [100000] * len(nums)
        dp[-1] = 0

        for i in range(len(nums) - 2, -1, -1):
            end = min(nums[i] + i + 1, len(nums))
            for j in range(i + 1, end):
                dp[i] = min(dp[i], dp[j] + 1)
        return dp[0]